### PR TITLE
Proposal to use explicit resources for service bindings

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
+
+## Context
+
+Service bindings are the configurations that determine how a Skupper
+service is invoked (ingress binding) and what processes that
+invocation may be routed to (egress binding).
+
+The Deployments or StatefulSets that are to be annotated may
+themselves by the generated output of other controllers. In this case
```
Comment (ajssmith) [inline doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:18]: typo? "themselves by"

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+by the skupper.io/address annotation.
+
+The code accummulated to handle all these configurations as well as
+the CLI is not the easiest to maintain.
+
+## Decision
+
+Explicit resources will be introduced that describe the desired
+configuration for both ingress- and egress- bindings.
+
```
Comment (ajssmith) [inline doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:48]: Suggest detailing the resource definition configuration (even if example only to be refined later)

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Date: 2024-01-16
+
+## Status
+
+Proposed
+
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
+
+## Context
+
```
Comment (pwright) [inline doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:12]: We decided that the user must explicitly create listeners  on any sites that require an exposed service, accepting that this is an extra step for users but overall it gives users more control.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Explicit resources will be introduced that describe the desired
+configuration for both ingress- and egress- bindings.
+
+## Consequences
+
+It is expected that a model made explicit in the resources used will
+be easier for users to understand and internalise, making it more
+likely that they are successful in their use of skupper and reducing
+their need to seek help with simple configuration.
+
```
Comment (pwright) [inline doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:55]:
```suggestion
The Skupper organisation needs to clearly communicate the changed model
to users, provide a path of existing users to migrate, and make sure new users are not confused by archaic references to ServiceSync.

```

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md (hunk 1 above)
Comment (grs) [reply doc/adr/0006-explicit-resources-for-declaring-service-bindings.md]: Yes, sorry, typo indeed. Fixed.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md (hunk 3 above)
Comment (grs) [reply doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:12]: That is really a consequence of the decision to remove service-sync which is described by adr 7 in which this point is now made.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md (hunk 4 above)
Comment (grs) [reply doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:55]: I've added words to that effect now.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md (hunk 2 above)
Comment (grs) [reply doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:48]: To my way of thinking, the details of the resource definitions to be used is a distinct decision from that of requiring explicit resources of some kind. I wrote this ADR with the view that a decision on the principle was important before trying to get a decision on the details and felt that including details would confuse that.

Happy to revisit that view if it is not shared, but just clarifying my thinking.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Proposed
+
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
+
+## Context
+
+Service bindings are the configurations that determine how a Skupper
+service is invoked (ingress binding) and what processes that
+invocation may be routed to (egress binding).
```
[the remark this answers is not in the document]
Comment (grs) [reply doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:15]: As mentioned in a comment above, I felt the detail of what these resources were to be called was a distinct decision from the principle of having explicit resources. In writing this I was really exploring the appropriate scope and granularity of decisions.

File: doc/adr/0006-explicit-resources-for-declaring-service-bindings.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+# 6. Explicit resources for declaring service bindings
+
+Date: 2024-01-16
+
+## Status
+
+Proposed
+
+Enables [7. Deprecate ServiceSync](0007-deprecate-servicesync.md)
```
Comment (fgiorgetti) [inline doc/adr/0006-explicit-resources-for-declaring-service-bindings.md:9]: I might be clear to state that the existing `skupper.io/proxy` annotation
is also going to be deprecated by this change.

**Review by nluaces — APPROVED** [review]
Looks good to me!

**Review by ajssmith — APPROVED** [review]

**Review by fgiorgetti — APPROVED** [review]

**Pull request merged**